mos_tests/murano/actions.py

```diff
@@ -0,0 +1,238 @@
+#    Copyright 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+import random
+import socket
+import telnetlib
+import time
+import yaml
+
+from mos_tests.functions.common import wait
+
+
+class MuranoActions(object):
+    """Murano-specific actions"""
+
+    def __init__(self, os_conn):
+        self.os_conn = os_conn
+
+    def rand_name(self, name):
+        return name + '_' + str(random.randint(1, 0x7fffffff))
+
+    def create_service(self, environment, session, json_data, to_json=True):
+        service = self.os_conn.murano.services.post(environment.id, path='/',
+                                                    data=json_data,
+                                                    session_id=session.id)
+        if to_json:
+            service = service.to_dict()
+            service = json.dumps(service)
+            return yaml.load(service)
+        else:
+            return service
+
+    def deploy_environment(self, environment, session):
+        self.os_conn.murano.sessions.deploy(environment.id, session.id)
+        try:
+            wait(lambda: self.os_conn.murano.environments.get(environment.id).
```

Comment:
Even if environment deploy failed, your code will wait ready status for 1000 seconds.
Let's rewrite it like below example

```python
def is_murano_env_deployed():
    status = self.os_conn.murano.environments.get(environment.id).status
    if status == 'error':
        raise Exception(...)
    return status == 'ready'

wait(is_murano_env_deployed, ...)
```